### DISCUSSION TITLE
Has there been a change to the locale feature of git for windows?

### DISCUSSION THREAD

**Discussion by sixwaaaay** [git-for-windows/git#4956] [Q&A]
I'm having problems updating from git version **v2.41.0.windows.1** to the latest **v2.45.1**.

In version 2.41, after I set LANG to zh_CN.utf-8, git's output works fine even though my powershell's code page is 936.
![image](https://github.com/git-for-windows/git/assets/56881699/46c5fb07-9315-42e1-a9c1-dfc61d22b611)

But in version **2.45.1**, using the original configuration resulted in garbled code, and I had to set both input and output encodings of console to utf-8 in order to get the git output to work properly.
```pwsh
[console]::InputEncoding = [System.Text.Encoding]::UTF8
[Console]::OutputEncoding = [System.Text.Encoding]::UTF8
```
Or if I set LANG=zh_CN.GBK, I can make git output work.
In short, following the settings from before the update doesn't work.
![image](https://github.com/git-for-windows/git/assets/56881699/852ab341-3702-4fad-847c-f14187b37d0b)

What's confusing is that  with `LANG=zh_CN.UTF-8`, I used `git log -h > 1.txt` to output to the file, and I found that both versions of the output were **utf-8** encoded.

----

pwsh version: PowerShell 7.4.2
windows version: 22631.3593

**Comment by dscho** [comment]
Please test the latest [snapshot](https://wingit.blob.core.windows.net/files/index.html).
